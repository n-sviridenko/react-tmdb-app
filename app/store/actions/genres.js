import { createAction } from 'redux-actions';

export const LIST_REQUEST = 'app/genres/LIST_REQUEST';
export const LIST_SUCCESS = 'app/genres/LIST_SUCCESS';
export const LIST_FAILURE = 'app/genres/LIST_FAILURE';

export const listRequest = createAction(LIST_REQUEST);
export const listSuccess = createAction(LIST_SUCCESS);
export const listFailure = createAction(LIST_FAILURE);
