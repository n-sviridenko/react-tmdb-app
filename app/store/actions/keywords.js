import { createAction } from 'redux-actions';

export const SEARCH_REQUEST = 'app/keywords/SEARCH_REQUEST';
export const SEARCH_SUCCESS = 'app/keywords/SEARCH_SUCCESS';
export const SEARCH_FAILURE = 'app/keywords/SEARCH_FAILURE';

export const searchRequest = createAction(SEARCH_REQUEST, null, (query) => ({ key: query }));
export const searchSuccess = createAction(SEARCH_SUCCESS, null, (res, key) => ({ key }));
export const searchFailure = createAction(SEARCH_FAILURE, null, (err, key) => ({ key }));

export const SHOW_REQUEST = 'app/keywords/SHOW_REQUEST';
export const SHOW_SUCCESS = 'app/keywords/SHOW_SUCCESS';
export const SHOW_FAILURE = 'app/keywords/SHOW_FAILURE';

export const showRequest = createAction(SHOW_REQUEST, null, (id) => ({ key: id }));
export const showSuccess = createAction(SHOW_SUCCESS, null, (res, key) => ({ key }));
export const showFailure = createAction(SHOW_FAILURE, null, (err, key) => ({ key }));
