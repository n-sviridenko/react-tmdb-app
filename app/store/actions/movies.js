import { createAction } from 'redux-actions';

export const DISCOVER_REQUEST = 'app/movies/DISCOVER_REQUEST';
export const DISCOVER_SUCCESS = 'app/movies/DISCOVER_SUCCESS';
export const DISCOVER_FAILURE = 'app/movies/DISCOVER_FAILURE';

export const discoverRequest = createAction(DISCOVER_REQUEST);
export const discoverSuccess = createAction(DISCOVER_SUCCESS);
export const discoverFailure = createAction(DISCOVER_FAILURE);
