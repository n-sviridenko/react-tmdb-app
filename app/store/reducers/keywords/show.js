import * as schemas from 'store/schemas';
import itemReducerCreator, { createRootSelector } from 'store/reducers/creators/loader';
import listReducerCreator, { createItemSelector } from 'store/reducers/creators/list';
import { SHOW_REQUEST, SHOW_SUCCESS, SHOW_FAILURE } from 'store/actions/keywords';

export default listReducerCreator({
  itemReducerCreator,
  types: [
    SHOW_REQUEST,
    SHOW_SUCCESS,
    SHOW_FAILURE,
  ],
  requestPayloadMapper: (data, id) => id,
});

const getRoot = (state) => state.getIn(['keywords', 'show']);
const makeGetRawById = (id) => createItemSelector(getRoot, id);
const makeGetById = (id) => createRootSelector(makeGetRawById(id), schemas.keyword);

export {
  makeGetById,
};
