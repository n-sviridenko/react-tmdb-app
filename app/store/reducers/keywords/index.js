import { combineReducers } from 'redux-immutable';

import search from './search';
import show from './show';

export default combineReducers({
  search,
  show,
});
