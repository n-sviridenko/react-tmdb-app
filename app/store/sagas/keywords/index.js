import { takeLatest, takeEvery } from 'redux-saga/effects';

import * as actions from 'store/actions/keywords';
import search from './search';
import show from './show';

export function* searchWatcher() {
  yield takeLatest(actions.SEARCH_REQUEST, search);
}

export function* showWatcher() {
  yield takeEvery(actions.SHOW_REQUEST, show);
}

export default [
  searchWatcher,
  showWatcher,
];
