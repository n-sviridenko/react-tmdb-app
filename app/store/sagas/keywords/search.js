import { call, put } from 'redux-saga/effects';

import * as schemas from 'store/schemas';
import * as actions from 'store/actions/keywords';
import { sendListRequest } from 'store/sagas/tmdbClient';

function* search({ payload: query, meta }) {
  try {
    const list = yield call(sendListRequest, 'search.getKeyword', schemas.keyword, { query });

    yield put(actions.searchSuccess(list, meta.key));
  } catch (err) {
    yield put(actions.searchFailure(err, meta.key));
  }
}

export default search;
