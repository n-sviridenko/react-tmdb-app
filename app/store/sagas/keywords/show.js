import { call, put } from 'redux-saga/effects';

import * as schemas from 'store/schemas';
import * as actions from 'store/actions/keywords';
import { sendRequest } from 'store/sagas/tmdbClient';

function* show({ payload: id, meta }) {
  try {
    const res = yield call(sendRequest, 'keywords.getById', schemas.keyword, { id });

    yield put(actions.showSuccess(res, meta.key));
  } catch (err) {
    yield put(actions.showFailure(err, meta.key));
  }
}

export default show;
