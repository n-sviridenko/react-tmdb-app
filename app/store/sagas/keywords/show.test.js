/* eslint-disable redux-saga/yield-effects */
import { put } from 'redux-saga/effects';

import * as actions from 'store/actions/keywords';
import show from './show';

describe('keywords show saga', () => {
  let task;

  beforeEach(() => {
    task = show({ payload: 1, meta: { key: 1 } });

    const callDescriptor = task.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the showSuccess if request was succeeded', () => {
    const putDescriptor = task.next(1).value;
    expect(putDescriptor).toEqual(put(actions.showSuccess(1, 1)));
  });

  it('should dispatch the showFailure if request was failed', () => {
    const error = new Error();
    const putDescriptor = task.throw(error).value;
    expect(putDescriptor).toEqual(put(actions.showFailure(error, 1)));
  });
});
