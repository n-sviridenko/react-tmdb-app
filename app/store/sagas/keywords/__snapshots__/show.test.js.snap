exports[`keywords show saga should dispatch the showFailure if request was failed 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "keywords.getById",
      EntitySchema {
        "_getId": [Function],
        "_idAttribute": "id",
        "_key": "keyword",
        "_mergeStrategy": [Function],
        "_processStrategy": [Function],
        "schema": Object {},
      },
      Object {
        "id": 1,
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`keywords show saga should dispatch the showSuccess if request was succeeded 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "keywords.getById",
      EntitySchema {
        "_getId": [Function],
        "_idAttribute": "id",
        "_key": "keyword",
        "_mergeStrategy": [Function],
        "_processStrategy": [Function],
        "schema": Object {},
      },
      Object {
        "id": 1,
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
